### ISSUE TITLE
[BUG-REPORT] Kernel dies for when reading beyond 4 million-th row of dataframe

### ISSUE THREAD

**Issue by hjazz6** [vaexio/vaex#1862]
**Description**
I was able to read multiple HDF5 files into a single vaex dataframe containing 1.12 billion rows. However, when I tried to do a `groupby` operation on the entire dataframe, the kernel dies.

I tried decreasing the size of the dataframe and found that when I'm trying to access beyond the 4 million-th row of the dataframe, the kernel usually dies. But the operation is always successful for the first 4 million rows.

```
# Executes successfully
df_short = df[:4000000]
df_count = df.groupby(df['columnA'], agg='count')
df.sort(by='count',ascending=False).head(10)

# Kernel dies
df_short = df[4000000:4500000]
df_count = df.groupby(df['columnA'], agg='count')
df.sort(by='count',ascending=False).head(10)
```

It does not seem related to the size of the dataframe, because the kernel fails when I'm trying to read 10 random rows beyond the 4 million-th row, e.g. `df[9001000:9001010]`, though there are a few times when reading some other 10 rows is successful. Basically, when reading beyond the 4 million-th row, the kernel dies about 70% of the time, but operation is always successful when reading up to the 4 million-th row.

**Software information**
 - Vaex version (`import vaex; vaex.__version__)`: 
vaex 4.7.0
vaex-astro 0.9.0
vaex-core 4.7.0.post1
vaex-hdf5 0.11.1
vaex-jupyter 0.7.0
vaex-ml 0.16.0
vaex-server 0.8.0
vaex-viz 0.5.1
vaex-graphql 0.2.0
 - Vaex was installed via: pip / conda-forge / from source 
 Offline install using `conda install` - first downloaded using `conda install` and `--download-only` on a PC, then transferred over to the current server and installed using `conda install vaex`. Additional packages were installed when the installation process complained that they were missing.
 - OS: RHEL 7.5

**Additional information**
I was unable to run `df['columnA'].nunique()` as the kernel died as well. But `columnA` should have fewer than 300 unique values, and is of format `str`.

**Comment by JovanVeljanoski** [comment]
Suggestion: can you try merge those files into a single massive hdf5 file and try the operations with that?
Basically open the data like you do now, and just immediately export it. 

In principle it should not make a difference (nowadays), but it is the first basic check.. 

We frequently test vaex on a 1.1 billion rows dataset to it is expected to work quite smoothly (but in our case it is a single hdf5 or arrow ) file.

**Comment by maartenbreddels** [comment]
Would be great if you can try to make a reproducible example, like generate some fake data.

**Comment by Ben-Epstein** [comment]
@maartenbreddels maybe this would help? https://github.com/vaexio/vaex/pull/1910

**Comment by hjazz6** [comment]
Sorry for the late reply, I was busy with something else.

@JovanVeljanoski I tried to merge and export it, but the kernel died. Shouldn't be a memory problem as there is still 88GB of free memory.

Here's what I tried:
```
all_files = glob.glob("/data/abc*.hdf5")
df = vaex.open_many(all_files)
df.export("/data1/merged.hdf5")  <-- kernel died while executing this
```

@maartenbreddels I'm not sure how I can reproduce it. I tried using the NYC taxi dataset (merging multiple CSVs together to get 12M records), but there was no problem with it. If it helps, in my dataset, `columnA` is a string, and there are a total of 29 columns in my dataset, 19 are `str`, 10 are `int64`. There are 321 HDF5 files totaling 371GB, and they are on a separate disk partition from where vaex/Jupyter notebook is running.

@Ben-Epstein Thanks. Is this function part of vaex yet? I was able to use the existing `open` instead of `open_many` to open my HDF5 files, though the kernel still died when trying to access beyond the 4M-th row.

**Comment by JovanVeljanoski** [comment]
Hmmm is the dtypes/schema of all those individual files the same? 
Some suggestions / ideas:
 - can you see if you have any file that has a string column for which the first element is either empty or missing or something like that (and leave them out during the concatenation/exporting)
 - my bet is that one (or more?) of those files is "special" i.e. causing the crash. Can you try to do it incrementally, say concat 50 files then another 50 and so on until you encounter the problem and try to isolate the file(s) that cause it. You can try with 2 or 3 files from the start. 

Basically we need more info to reproduce this issue so we can fix it..
